charts/redpanda/chart_test.go

```diff
@@ -200,6 +202,114 @@ func TestChart(t *testing.T) {
 	})
 }
 
+func TestConfigMap(t *testing.T) {
+	ctx := testutil.Context(t)
+	client, err := helm.New(helm.Options{ConfigHome: testutil.TempDir(t)})
+	require.NoError(t, err)
+
+	// Chart deps are kept within ./charts as a tgz archive, which is git
+	// ignored. Helm dep build will ensure that ./charts is in sync with
+	// Chart.lock, which is tracked by git.
+	require.NoError(t, client.RepoAdd(ctx, "redpanda", "https://charts.redpanda.com"))
+	require.NoError(t, client.DependencyBuild(ctx, "."), "failed to refresh helm dependencies")
+
+	values, err := os.ReadDir("./ci")
+	require.NoError(t, err)
+
+	for _, v := range values {
+		v := v
+		t.Run(v.Name(), func(t *testing.T) {
+			t.Parallel()
+
+			manifests, err := client.Template(ctx, "redpanda/redpanda", helm.TemplateOptions{
+				Name:       "redpanda",
+				ValuesFile: "./ci/" + v.Name(),
+				Set: []string{
+					// Tests utilize some non-deterministic helpers (rng). We don't
+					// really care about the stability of their output, so globally
+					// disable them.
+					"tests.enabled=false",
+					// jwtSecret defaults to a random string. Can't have that
+					// in snapshot testing so set it to a static value.
+					"console.secret.login.jwtSecret=SECRETKEY",
+				},
+				Version: "5.8.5",
+			})
+			require.NoError(t, err)
+
+			oldConf, err := getRedpandaConfiguration(manifests)
+			require.NoError(t, err)
+
+			manifests, err = client.Template(ctx, ".", helm.TemplateOptions{
+				Name:       "redpanda",
+				ValuesFile: "./ci/" + v.Name(),
+				Set: []string{
+					// Tests utilize some non-deterministic helpers (rng). We don't
+					// really care about the stability of their output, so globally
+					// disable them.
+					"tests.enabled=false",
+					// jwtSecret defaults to a random string. Can't have that
+					// in snapshot testing so set it to a static value.
+					"console.secret.login.jwtSecret=SECRETKEY",
+				},
+			})
+			require.NoError(t, err)
+
+			newConf, err := getRedpandaConfiguration(manifests)
+			require.NoError(t, err)
+
+			require.JSONEq(t, oldConf.redpanda, newConf.redpanda)
+			require.JSONEq(t, oldConf.bootstrap, newConf.bootstrap)
+			require.JSONEq(t, oldConf.rpkProfile, newConf.rpkProfile)
+		})
+	}
+}
+
+type Configuration struct {
+	redpanda   string
+	bootstrap  string
+	rpkProfile string
+}
+
+func getRedpandaConfiguration(manifests []byte) (*Configuration, error) {
```

Comment:
nit: Add a small comment here to disambiguate what `RedpandaConfiguration` actually is.